src/dialects/ibmi/data-types.js

```diff
@@ -0,0 +1,331 @@
+      warn(`IBM i Db2 does not support '${dataType.key}' with LENGTH, UNSIGNED or ZEROFILL. Plain '${dataType.key}' will be used instead.`);
+      dataType._length = undefined;
+      dataType.options.length = undefined;
+      dataType._unsigned = undefined;
+      dataType._zerofill = undefined;
+    }
+  }
+
+  /**
+   * @see https://www.ibm.com/support/knowledgecenter/en/ssw_ibm_i_73/db2/rbafzch2data.htm
+   */
+
+  BaseTypes.DATE.types.ibmi = [93];
+  BaseTypes.STRING.types.ibmi = [12];
+  BaseTypes.CHAR.types.ibmi = [1];
+  BaseTypes.TEXT.types.ibmi = [12];
+  BaseTypes.TINYINT.types.ibmi = false;
+  BaseTypes.SMALLINT.types.ibmi = [5];
+  BaseTypes.MEDIUMINT.types.ibmi = false;
+  BaseTypes.INTEGER.types.ibmi = [4];
+  BaseTypes.BIGINT.types.ibmi = [-5];
+  BaseTypes.FLOAT.types.ibmi = [8];
+  BaseTypes.TIME.types.ibmi = [92];
+  BaseTypes.DATEONLY.types.ibmi = [91];
+  BaseTypes.BOOLEAN.types.ibmi = [5];
+  BaseTypes.BLOB.types.ibmi = [-4, -2];
+  BaseTypes.DECIMAL.types.ibmi = [3];
+  BaseTypes.UUID.types.ibmi = [1];
+  BaseTypes.ENUM.types.ibmi = false;
+  BaseTypes.REAL.types.ibmi = [7];
+  BaseTypes.DOUBLE.types.ibmi = [8];
+  BaseTypes.GEOMETRY.types.ibmi = false;
+  BaseTypes.JSON.types.ibmi = false;
+
+  class ENUM extends BaseTypes.ENUM {
+    toSql() {
+      return 'VARCHAR(255)';
+    }
+  }
+
+  class DATE extends BaseTypes.DATE {
+    toSql() {
+      return 'TIMESTAMP';
+    }
+
+    static parse(date) {
+      if (!date.includes('+')) {
+        // For backwards compat. Dates inserted by sequelize < 2.0dev12 will not have a timestamp set
+        const mome = moment.utc(date);
+
+        return mome.toDate();
+      }
+
+      const mome = moment.utc(date);
+
+      return mome.toDate();
+    }
+
+    _stringify(date, options) {
+      date = this._applyTimezone(date, options);
+
+      return date.format('YYYY-MM-DD HH:mm:ss.SSS');
+    }
+  }
+
+  class UUID extends BaseTypes.UUID {
+    toSql() {
+      return 'CHAR(36)';
+    }
+  }
+
+  class DATEONLY extends BaseTypes.DATEONLY {
+    static parse(date) {
+      return date;
+    }
+  }
+
+  class BOOLEAN extends BaseTypes.BOOLEAN {
+    toSql() {
+      return 'SMALLINT';
+    }
+
+    _stringify(value) {
+      if (value) {
+        return 1;
+      }
+
+      return 0;
+    }
+
+    static parse(value) {
+      if (value) {
+        return true;
+      }
+
+      return false;
+    }
+  }
+
+  class STRING extends BaseTypes.STRING {
+    toSql() {
+      if (this._binary) {
+        return `BINARY(${this._length})`;
+      }
+
+      return super.toSql(this);
+    }
+
+    _stringify(value, options) {
+      if (Buffer.isBuffer(value)) {
+        return `BLOB(X'${value.toString('hex')}')`;
+      }
+
+      return value.toString();
+    }
+
+  }
+
+  class TEXT extends BaseTypes.TEXT {
+    toSql() {
+      let len = 0;
+      if (this._length) {
+        if (isNaN(this._length)) {
```

Comment:
using `isNaN` is banned by our linter (but not run on js files, only ts, as we resolve these issues one by one) because it's a code-smell.

If you want to detect whether something is NaN, use Number.isNaN
If you're trying to detect whether this._length is a number, use `Number.isSafeInteger`
If it's not a safe integer, and it's not one of the 3 accepted strings, and it's not undefined (for default value), I think this should throw.